explorer/deploy/webAppCreator.ts

```diff
@@ -172,9 +173,16 @@ class ResourceGroupStep extends WebAppCreatorStepBase {
             resourceGroups = results[0];
             locations = results[1];
             resourceGroups.forEach(rg => {
+                let locationDisplayName: string;
+                try {
+                    locationDisplayName = locations.find(l => l.name.toLowerCase() === rg.location.toLowerCase()).displayName;
+                } catch (error) {
```

Comment:
What error is occurring?  Null ref?  I'd rather avoid exceptions than catch and recover.

Reply:
Exactly, I could also store the result in a variable and check if it is available before assigning the value, say : 
const location : Location = locations.find(l => l.name.toLowerCase() === rg.location.toLowerCase());

if(location){
 locationDisplayName = location.displayName;
}else{
 ...
 }

Alternatively I could try changing the location acquisition by subscription strategy and do so by overall locations which would prevent the error in the grand majority of the very rare cases although I suspect this would have problems in cases when there is a private azure location. Also, is there a specific reason as to prefer avoiding rather than catch and recover?

Reply:
Yes, that's preferable, thanks.  Exceptions should be for things that are exceptional.  Really you should never see a null ref exception, it's completely avoidable with proper checking.  Having the location not match is a fully normal event that we just need to handle.

(Also, having a bunch of exceptions throw all over the place is darn annoying when you're trying to debug.)

Reply:
Sounds good, I'll try to follow that paradigm.